### ISSUE TITLE
Dummy provisioner for testing purpose

### ISSUE THREAD

**Issue by dav-m85** [lafourchette/prototype#5]
Testing with the current Provisioner/Vagrant is quite hard, you need a box and need to wait ages while provisionning. Quite a pain just to test the interface when debugging.

It would be nice to have a Dummy provisionner, which supports all start/stop commands, but that does NOT actually creates a VM. I think tracking a text file would be enough.

**Comment by suarezd** [comment]
Hi all ! Thanks for giving me the opportunity !

**Comment by suarezd** [comment]
Hi @dav-m85, i tried to run PHPUnit tests but it looks like i'm missing something.
I already ran composer install but even if i execute tests file by file it doesn't find some Classes path.
Any Idea please ?
Should i push my PHPUnit tests anyway so you can look at it or it doesn't work like this ? :)
Thank you for the help.

**Comment by dav-m85** [comment]
@suarezd You are right, the tests are not loading on my environment too. Are you up for fixing them ? It is not part of the issue but I'll definitively love them running (no need to make them green, just running would be fine).

**Comment by suarezd** [comment]
Sure, I've changed the path in the phpunit.xml.dist file and they're running again.

**Comment by suarezd** [comment]
Just in case, you'll see in the first commit of mine a wrong email, i was at the office and i forgot to change the git config user.email. Sorry.

**Comment by dav-m85** [comment]
git rebase --amend is your friend ;)

**Comment by suarezd** [comment]
@dav-m85 i've tried to fork prototype to my repos but everytime i try to push i have a fatal error authentication failed. Do I need to be contributor in order to push ?

**Comment by dav-m85** [comment]
You have push access to your fork, aka suarezd/prototype. You have to be collaborator to push to lafourchette/prototype indeed. That's the purpose of pull requests!

**Comment by suarezd** [comment]
I've just submited my pull request. thanks a lot for the help @dav-m85 !

**Comment by dav-m85** [comment]
Testing with the current ```Provisioner/Vagrant``` is quite a pain in the ass, you need a box and need to wait ages while provisionning. Quite a hurdle just to test the interface when debugging for example.

Your mission Jim, should you decide to accept it, is to:
- Install the project on your machine and build it (see README.md)
- Create a new Provisioner, that instead of creating a VagrantFile and ```vagrant up``` it, just stores some data in a file on the target installation directory.
- Of course, this DummyProvisioner shall support all commands from the ```ProvisionerInterface```.
- Make this new DummyProvisioner the default option in config.json for new installations.
- Tests are appreciated.

You can have a look at ```Provisioner/Vagrant::generateVagrantfile()``` as a start. Ow, by the way, current node status are updated by cron. you have the ```dist/cron.conf``` file if you need it.

This tape will NOT self-destruct in five seconds. Good luck, Jim.

**Comment by dav-m85** [comment]
Your mission Jim, should you decide to accept it, is to:

Install the project on your machine and build it (see README.md)
Create a new Provisioner, that instead of creating a VagrantFile and vagrant up it, just stores some data in a file on the target installation directory.
Of course, this DummyProvisioner shall support all commands from the ProvisionerInterface.
Make this new DummyProvisioner the default option in config.json for new installations.
Tests are appreciated.
You can have a look at Provisioner/Vagrant::generateVagrantfile() as a start. Ow, by the way, current node status are updated by cron. you have the dist/cron.conf file if you need it.

This tape will NOT self-destruct in five seconds. Good luck, Jim.
